dask/_task_spec.py

```diff
@@ -692,6 +692,32 @@ def is_coro(self):
                 self._is_coro = False
         return self._is_coro
 
+    @staticmethod
+    def fuse(*tasks: Task, key: KeyType | None = None) -> Task:
+        leafs = set()
+        all_keys = set()
+        all_deps: set[KeyType] = set()
+        for t in tasks:
+            if t.key not in all_deps:
+                leafs.add(t.key)
+            all_deps.update(t.dependencies)
+            all_keys.add(t.key)
+            leafs -= t.dependencies
```

Comment:
Does this rely on the order in which we iterate over tasks?

It might be easier to define leafs `all_keys - all_deps`.

Reply:
not the same thing

Reply:
> Does this rely on the order in which we iterate over tasks?

The couple tests I added are testing permutations

Reply:
> not the same thing

Could you elaborate on what `leafs` are supposed to be? I don't think that I'm following then.

Reply:
For future reference: This is _not_ order-dependent because a non-leaf task is either added before its dependent, in which case it will be removed via `leafs -= t.dependencies` when the dependent is added, or it's added after its dependent, in which case its already in `all_deps`.

Reply:
sorry. Leafs/roots may be confusing terminology here. This is identifying the task that will ultimately provide the final data. All other tasks in here will be direct or transitive dependencies. If the later is not the case, the fused task is invalid and this will raise.

Reply:
> sorry. Leafs/roots may be confusing terminology here. This is identifying the task that will ultimately provide the final data. All other tasks in here will be direct or transitive dependencies. If the later is not the case, the fused task is invalid and this will raise.

Isn't that exactly what `all_keys` (all the tasks in this subset) `- all_deps` (except for the ones that are dependencies of any task in this subset) would return, or am I missing some subtlety here? (This should be calculated after the loop here.)